geth/node/status_node.go

```diff
@@ -135,44 +152,56 @@ func (n *StatusNode) startPeerPool() error {
 func (n *StatusNode) Stop() error {
 	n.mu.Lock()
 	defer n.mu.Unlock()
+
+	if !n.isRunning() {
+		return ErrNoRunningNode
+	}
+
 	return n.stop()
 }
 
 // stop will stop current StatusNode. A stopped node cannot be resumed.
 func (n *StatusNode) stop() error {
-	if err := n.isAvailable(); err != nil {
-		return err
-	}
-	if n.gethNode.Server().DiscV5 != nil {
-		n.stopPeerPool()
+	if err := n.stopPeerPool(); err != nil {
+		n.log.Error("Error stopping the PeerPool", "error", err)
 	}
+	n.register = nil
+	n.peerPool = nil
+	n.db = nil
+
 	if err := n.gethNode.Stop(); err != nil {
 		return err
 	}
-	n.gethNode = nil
-	n.config = nil
+
```

Comment:
In case the node is not ephimeral you also should reset the config here?

Reply:
Hm, I guess I should keep it the same as it was before because I do not have a good argument to keep it after the node is stopped.